### ISSUE TITLE
Wrong OpenMP linking with ifort

### ISSUE THREAD

**Issue by uihsnv** [miroi/lapack-dsyevr-test#1]
python setup --fc=ifort --cmake-options="-D MATH_LIB_SEARCH_ORDER='MKL'"  build_ifort_mkl

generates the wrong OpenMP link options in the file: 

    ./build_ifort_mkl/src/CMakeFiles/dsyevr_check.dir/link.txt

It should be using *-qopenmp*, because I'm using an intel compiler of version 18.0.

This looks an issue with **autocmake**.  The following file should take care of that, but is there some reason it isn't?

https://github.com/coderefinery/autocmake/blob/master/modules/omp.cmake

**Comment by miroi** [comment]
Hi,
I haven't checked it yet, maybe it's an autocmake issue. BTW, Rado ( @bast ) , the macro https://github.com/coderefinery/autocmake/blob/master/modules/omp.cmake should properly identify Intel version 18 ? 

Hmm, looking into  https://github.com/coderefinery/autocmake/blob/master/modules/omp.cmake#L54  is the CMake variable "${CMAKE_Fortran_COMPILER_VERSION}" always properly filled ?  CMake identification of Fortran compilers version does not always work, that's why we have python macro for doing that.

**Comment by bast** [comment]
What CMake version are you using?

**Comment by uihsnv** [comment]
@bast 3.10.3

**Comment by bast** [comment]
And what flags do you get?

**Comment by uihsnv** [comment]
The other one for the older versions of the Intel compiler: *-openmp*

@miroi btw, what exactly does this bug affect? Are eigenvalues unaffected? Is it only the eigenvectors? And are all eigenvectors wrong?

**Comment by bast** [comment]
What confuses me here is that the OMP module is not used at all in this project. But perhaps you have added it in some other fork or local repository?

**Comment by bast** [comment]
Also I cannot reproduce it. On CMake above 3.4, it should not go into any of the ifs since it has support for Fortran OMP flags.

**Comment by miroi** [comment]
@uihsnv  see updated https://github.com/miroi/lapack-dsyevr-test/blob/master/README.rst . This bug affects eigenvectors; IMO, each DSYERV implementation should be checked.

**Comment by uihsnv** [comment]
@bast Nope. I made no changes to this; I just cloned it and ran it as described in the README.
